spec/requests/conversion_hosts_spec.rb

```diff
@@ -45,4 +45,198 @@
       expect(response).to have_http_status(:forbidden)
     end
   end
+
+  context "create" do
+    let(:vm) { FactoryGirl.create(:vm) }
+    let(:host) { FactoryGirl.create(:host) }
+
+    let(:sample_conversion_host_from_vm) do
+      {
+        :name          => "test_conversion_host_from_vm",
+        :resource_type => "vm",
+        :resource_id   => vm.id,
+        :version       => "1.0"
+      }
+    end
+
+    let(:sample_conversion_host_from_host) do
+      {
+        :name          => "test_conversion_host_from_host",
+        :resource_type => "host",
+        :resource_id   => host.id,
+        :version       => "1.0"
+      }
+    end
+
+    let(:expected_attributes) { %w(id name resource_type resource_id version) }
+
+    it "supports single conversion host creation" do
+      api_basic_authorize(collection_action_identifier(:conversion_hosts, :create))
+
+      post(api_conversion_hosts_url, :params => sample_conversion_host_from_vm)
+
+      expect(response).to have_http_status(:ok)
+      expect_result_resources_to_include_keys("results", expected_attributes)
+
+      conversion_host_id = response.parsed_body["results"].first["id"]
+      expect(ConversionHost.exists?(conversion_host_id)).to be_truthy
+    end
+
+    it "supports single conversion host creation via action" do
+      api_basic_authorize(collection_action_identifier(:conversion_hosts, :create))
+
+      post(api_conversion_hosts_url, :params => gen_request(:create, sample_conversion_host_from_vm))
+
+      expect(response).to have_http_status(:ok)
+      expect_result_resources_to_include_keys("results", expected_attributes)
+
+      conversion_host_id = response.parsed_body["results"].first["id"]
+      expect(ConversionHost.exists?(conversion_host_id)).to be_truthy
+    end
+
+    it "supports multiple conversion host creation" do
+      api_basic_authorize(collection_action_identifier(:conversion_hosts, :create))
+
+      conversion_hosts = [sample_conversion_host_from_vm, sample_conversion_host_from_host]
+      post(api_conversion_hosts_url, :params => gen_request(:create, conversion_hosts))
+
+      expect(response).to have_http_status(:ok)
+      expect_result_resources_to_include_keys("results", expected_attributes)
+
+      results = response.parsed_body["results"]
+
+      expect(ConversionHost.exists?(results.first["id"])).to be_truthy
+      expect(ConversionHost.exists?(results.last["id"])).to be_truthy
+      expect(results.first["resource_id"]).to eq(vm.id.to_s)
```

Comment:
just curious, why the returned `id` a string?  Is the body a json?

Reply:
Yes, the body is json (and `parsed_body` is a Ruby hash of that).